### ISSUE TITLE
Distrobox always fails

### ISSUE THREAD

**Issue by Locutus64** [topgrade-rs/topgrade#92]
I installed Garuda yesterday and have been slowly building it up. Shortly after installing I installed you version of Topgrade. Everytime it's run it gets to a tool called Distrobox and won't do anything with it, it just fails and I'm asked if I want to try again. I looked for Distrobox using Pamac and of the 3 versions listed according to Pamac none are installed. I contacted Garuda to see if it was something they added and according to them no.  I'm at a lost on how to either fix Distrobox or remove it since it's not liusted as installed. Any help with this would be greatly appreciated. Thanks

## Additional Details
- Garuda - Arch based distro
- Use pamac to install Topgrade - it just updated
- Topgrade version 10.0.1

```
╭─locutus@Unimatrix in ~ 
╰─λ topgrade

―― 07:06:53 - System update ――――――――――――――――――――――――――――――――――――――――――――――――――――
[sudo] password for locutus:
:: Synchronizing package databases...
garuda is up to date
core is up to date
extra is up to date
community is up to date
multilib is up to date
chaotic-aur                              2.3 MiB  4.89 MiB/s 00:00 [-------------------------------------] 100%
arcolinux_repo                         131.7 KiB   775 KiB/s 00:00 [-------------------------------------] 100%
arcolinux_repo_3party                  283.3 KiB  1445 KiB/s 00:00 [-------------------------------------] 100%
arcolinux_repo_xlarge                   34.6 KiB   324 KiB/s 00:00 [-------------------------------------] 100%
nemesis_repo is up to date
endeavouros is up to date
xerolinux_repo is up to date
xerolinux_repo_xl is up to date
:: Starting full system upgrade...
warning: topgrade-git: local (10.0.1.r1163.5a6f0d2-1) is newer than arcolinux_repo_3party (r1080.57f38a0-1)
there is nothing to do
:: Searching databases for updates...
:: Searching AUR for updates...
-> topgrade-git: local (10.0.1.r1163.5a6f0d2-1) is newer than arcolinux_repo_3party (r1080.57f38a0-1)
-> Flagged Out Of Date AUR Packages:  komga
there is nothing to do

―― 07:07:05 - Configuration update ―――――――――――――――――――――――――――――――――――――――――――――
==> pacnew file found for /etc/pamac.conf
:: (V)iew, (M)erge, (S)kip, (R)emove pacnew, (O)verwrite with pacnew, (Q)uit: [v/m/s/r/o/q] s

―― 07:07:09 - oh-my-zsh ――――――――――――――――――――――――――――――――――――――――――――――――――――――――
Pulling custom plugins and themes
Pulling /home/locutus/.oh-my-zsh/custom/plugins/zsh-syntax-highlighting
Pulling /home/locutus/.oh-my-zsh/custom/themes/powerlevel10k
Pulling /home/locutus/.oh-my-zsh/custom/plugins/zsh-autosuggestions
Pulling /home/locutus/.oh-my-zsh/custom/plugins/zsh-completions
Up-to-date /home/locutus/.oh-my-zsh/custom/themes/powerlevel10k
Up-to-date /home/locutus/.oh-my-zsh/custom/plugins/zsh-completions
Up-to-date /home/locutus/.oh-my-zsh/custom/plugins/zsh-syntax-highlighting
Up-to-date /home/locutus/.oh-my-zsh/custom/plugins/zsh-autosuggestions
Updating Oh My Zsh
__                                     __
____  / /_     ____ ___  __  __   ____  _____/ /_
/ __ \/ __ \   / __ `__ \/ / / /  /_  / / ___/ __ \
/ /_/ / / / /  / / / / / / /_/ /    / /_(__  ) / / /
\____/_/ /_/  /_/ /_/ /_/\__, /    /___/____/_/ /_/
/____/

Oh My Zsh is already at the latest version.

To keep up with the latest news and updates, follow us on Twitter: https://twitter.com/ohmyzsh
Want to get involved in the community? Join our Discord: https://discord.gg/ohmyzsh
Get your Oh My Zsh swag at: https://shop.planetargon.com/collections/oh-my-zsh

―― 07:07:10 - TLDR ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
Successfully updated cache.

―― 07:07:10 - micro ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
Checking for plugin updates
Nothing to install / update

―― 07:07:11 - Flatpak User Packages ――――――――――――――――――――――――――――――――――――――――――――
Looking for updates…

Nothing to do.

―― 07:07:11 - Flatpak System Packages ――――――――――――――――――――――――――――――――――――――――――
Looking for updates…

Nothing to do.

―― 07:07:11 - Distrobox ――――――――――――――――――――――――――――――――――――――――――――――――――――――――

Retry? (y)es/(N)o/(s)hell/(q)uit

―― 07:07:13 - Firmware upgrades ――――――――――――――――――――――――――――――――――――――――――――――――
Firmware metadata last refresh: 18 hours ago. Use --force to refresh again.
Devices with no available firmware updates:
• System Firmware
• Unifying Receiver
• Unifying Receiver
• WD Blue SN570 500GB
Devices with the latest available firmware version:
• UEFI dbx
No updates available

―― 07:07:13 - Summary ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
System update: OK
config-update: OK
oh-my-zsh: OK
TLDR: OK
micro: OK
Flatpak: OK
distrobox: FAILED
Firmware upgrades: OK

Pacman backup configuration files found:
/etc/pamac.conf.pacnew
/etc/paru.conf.pacsave
```

```
╭─locutus@Unimatrix in ~ as 🧙 took 1ms
╰─λ topgrade --dry-run

―― 07:09:36 - System update ――――――――――――――――――――――――――――――――――――――――――――――――――――
Dry running: /usr/bin/yay --pacman pacman -Syu

―― 07:09:37 - Configuration update ―――――――――――――――――――――――――――――――――――――――――――――
Dry running: /usr/bin/sudo --preserve-env=DIFFPROG /usr/bin/pacdiff

―― 07:09:37 - oh-my-zsh ――――――――――――――――――――――――――――――――――――――――――――――――――――――――
Pulling custom plugins and themes
Would pull /home/locutus/.oh-my-zsh/custom/plugins/zsh-autosuggestions
Would pull /home/locutus/.oh-my-zsh/custom/plugins/zsh-syntax-highlighting
Would pull /home/locutus/.oh-my-zsh/custom/plugins/zsh-completions
Would pull /home/locutus/.oh-my-zsh/custom/themes/powerlevel10k
Dry running: zsh /home/locutus/.oh-my-zsh/tools/upgrade.sh

―― 07:09:37 - TLDR ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
Dry running: /usr/bin/tldr --update

―― 07:09:37 - micro ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
Dry running: /usr/bin/micro -plugin update

―― 07:09:37 - Flatpak User Packages ――――――――――――――――――――――――――――――――――――――――――――
Dry running: /usr/bin/flatpak update --user

―― 07:09:37 - Flatpak System Packages ――――――――――――――――――――――――――――――――――――――――――
Dry running: /usr/bin/flatpak update --system

―― 07:09:37 - Distrobox ――――――――――――――――――――――――――――――――――――――――――――――――――――――――
Dry running: distrobox upgrade --all

―― 07:09:37 - Firmware upgrades ――――――――――――――――――――――――――――――――――――――――――――――――
Dry running: /usr/bin/fwupdmgr refresh
Dry running: /usr/bin/fwupdmgr get-updates

―― 07:09:37 - Summary ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
System update: OK
config-update: OK
oh-my-zsh: OK
TLDR: OK
Flatpak: OK
distrobox: OK
Firmware upgrades: OK

Pacman backup configuration files found:
/etc/pamac.conf.pacnew
/etc/paru.conf.pacsave
```

**Comment by Locutus64** [comment]
OK a member of the Garuda team found the below. Now how to disable the Distrobox check? Thanks

https://github.com/topgrade-rs/topgrade/pull/76

**Comment by Locutus64** [comment]
With help from TechXero I got the topgrade.toml line 5 uncommented and edited to. Works perfectly now.

disable = ["emacs", "distrobox"]

**Comment by DottoDev** [comment]
Great to hear that it works now.

**Comment by Locutus64** [comment]
> Great to hear that it works now.

I'm assuming it's a bug since since distrobox should be poled by topgrade since I don't have it installed.

**Comment by DottoDev** [comment]
I don't know why it's run even when you don't have it installed, but I will look into that.

**Comment by yochananmarqos** [comment]
Here's the verbose output from 10.1.0:

```
― 11:57:02 - Distrobox ――――――――――――――――――――――――――――――――――――――――――――――――――――――――
 2022-10-31T17:57:02.347Z DEBUG topgrade::executor        > Running "distrobox" "upgrade" "--all"
 2022-10-31T17:57:02.347Z DEBUG topgrade::runner          > Step "distrobox" failed: No such file or directory (os error 2)
 2022-10-31T17:57:02.347Z DEBUG topgrade::terminal        > Desktop notification: distrobox failed
```

**Comment by Locutus64** [comment]
XeroTech installed this version and got the same issue, so it's not a one off.

**Comment by DottoDev** [comment]
Due to a big it seems like distrobox gets always invoked, not only when it's installed/required.

**Comment by DottoDev** [comment]
Can anyone check if this occures using the version from the bug-fixes branch.

**Comment by jojojojojoj5564656465465** [comment]
Hello i have almost same bug  

`disable = [ "distrobox", "git-repo"]`

Unfortunatly topgrade keep trying to do the updates and fails.
I am using your software from AUR

**Comment by DottoDev** [comment]
> Hello i have almost same bug  
> 
> 
> 
> `disable = [ "distrobox", "git-repo"]`
> 
> 
> 
> Unfortunatly topgrade keep trying to do the updates and fails.
> 
> I am using your software from AUR

Could you do the following?

clone the git repo
Checkout the bug-fixes branch
Run topgrade with the following command: `cargo run --release`

**Comment by jojojojojoj5564656465465** [comment]
thanks i think it's because i didn't add `${XDG_CONFIG_HOME:-~/.config}/topgrade.toml`

I added the line like this in my zshrc and my terminal open with an error

`[WARNING]: Console output during zsh initialization detected.

When using Powerlevel10k with instant prompt, console output during zsh
initialization may indicate issues.

You can:

  - Recommended: Change ~/.zshrc so that it does not perform console I/O
    after the instant prompt preamble. See the link below for details.

    * You will not see this error message again.
    * Zsh will start quickly and prompt will update smoothly.

  - Suppress this warning either by running p10k configure or by manually
    defining the following parameter:

      typeset -g POWERLEVEL9K_INSTANT_PROMPT=quiet

    * You will not see this error message again.
    * Zsh will start quickly but prompt will jump down after initialization.

  - Disable instant prompt either by running p10k configure or by manually
    defining the following parameter:

      typeset -g POWERLEVEL9K_INSTANT_PROMPT=off

    * You will not see this error message again.
    * Zsh will start slowly.

  - Do nothing.

    * You will see this error message every time you start zsh.
    * Zsh will start quickly but prompt will jump down after initialization.

For details, see:
https://github.com/romkatv/powerlevel10k/blob/master/README.md#instant-prompt

-- console output produced during zsh initialization follows --

/home/endevour/.zshrc:215: permission not alowed: /home/endevour/.config/topgrade.toml`

**Comment by DottoDev** [comment]
If you put your topgrade.toml file into the .config folder this should work out of the box without adding anything to your .zshrc file.

**Comment by jojojojojoj5564656465465** [comment]
Do you have a tool to build from source ?

**Comment by DottoDev** [comment]
You can build from source using git and cargo.

**Comment by Samisafool** [comment]
Please fix this. I guess it's something in the detection code whether to run the step or not, which always passes, and if no distrobox then the step fails

**Comment by DottoDev** [comment]
It could be that it's fixed with the latest release, but I will look into it.
